routes/variableurl.go

```diff
@@ -0,0 +1,123 @@
+package routes
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type (
+	parseIndex uint
+
+	ArgName string
```

Comment:
I don’t think requiring users to specify `routes.ArgName("arg")` when using the decorator is the way to go : it is very legible and does not provide any new info. Perhaps you should redefine your structures to use a simple `string` and convert them to `ArgName` behind the scene, that way `routes.ParseInt{routes.ArgName("arg")}` becomes `routes.ParseInt{"arg"}`.